envs/vpn/conda/fedbiomed-node.yaml

```diff
@@ -10,56 +10,55 @@ channels:
 
 dependencies:
   # minimal environment
-  - python >=3.9,<3.10
-  - pip
-  - jupyter
-  - ipython
+  # python 3.11 recently released 2022-11 and not yet supported by some deps including: torchvision
+  - python >=3.10,<3.11
+  - pip >= 23.0
+  - jupyter ~=1.0.0
+  - ipython ~=8.13.2
   # tests
-  - pytest >6.2.2
-  - tinydb >=4.4.0,<5.0.0
-  - tabulate >=0.8.9,<0.9.0
+  - pytest ~=7.2.0
+  - pytest-cov ~=4.1.0
+  - tinydb ~=4.7.1
+  - tabulate >=0.9.0,<0.10.0
   # code
-  - GitPython >=3.1.14,<4.0.0
-  - requests >=2.25.1,<3.0.0
-  - paho-mqtt >=1.5.1,<2.0.0
-  - validators >=0.18.2,<0.19.0
-  - tqdm >=4.59.0,<5.0.0
-  - git
-  - packaging >=23.0,<24.0
+  - requests ~=2.29.0
+  - paho-mqtt ~=1.6.1
+  - validators >=0.20.0,<0.21.0
```

Comment:
I saw that this module is used only for validating URLs. I think a regex implementation will be enough to validate a URL or built-in `urllib`. Using an  third-party module seemed unnecessary to me.  However, it is not in the scope of this PR. I am leaving this comment to leave a track. I may create a new issue for that.

Reply:
Thanks for noticing.
I would favor using `urllib` over custom regexp.
I suspect the initial idea was that we would have to check many types of data, but as it is not the case, it is probably better to stick to a simpler alternative, as you suggest.